### PR TITLE
Only allow latin-1 encodable characters in username and password

### PR DESCRIPTION
**Pull request by i-oden**
## **1. This PR contains the following changes...**

See https://github.com/ScilifelabDataCentre/dds_cli/pull/617 for background.

This adds a new validation to make sure that accepted usernames and passwords are able to be passed through the requests package from the CLI.

## **2. The following additional changes are required for this to work**

_X_

## **3. The PR fixes the following GitHub issue / Jira task**

- [ ] GitHub issue (link):
- [x] Jira task (ID, `DDS-xxxx`): DDS-1489
- [ ] The PR does not fix a specific GitHub issue or Jira task

## **4. What _type of change(s)_ does the PR contain?**

- [ ] New feature
  - [ ] Breaking: _Please describe the reason for the break and how we can fix it._
  - [ ] Non-breaking
- [ ] Database change
  - [ ] Migration _included in PR_
  - [ ] Migration _not needed_
- [x] Bug fix
  - [x] Breaking: Not **breaking** but some users could potentially need to reset their passwords. Not likely though, in this case they'd have contacted us about the UnicodeEncodeError.
  - [ ] Non-breaking
- [ ] Security Alert fix
- [ ] Documentation
- [ ] Tests **(only)**
- [ ] Workflow

### **Always**

- [Changelog](../CHANGELOG.md)
  - [x] Added
  - [ ] Not needed (E.g. PR contains _only_ tests)
- Rebase / Update / Merge _from_ base branch (the branch from which the current is forked)
  - [ ] Done
  - [x] Not needed
- Blocking PRs
  - [ ] Merged
  - [x] No blocking PRs
- PR to `master` branch
  - [ ] Yes: Go to the section [PR to master](#pr-to-master)
  - [x] No

### If PR consists of **code change(s)**

- Self review
  - [x] Done
- Comments, docstrings, etc
  - [x] Added / Updated
- Documentation
  - [ ] Updated
  - [x] Update not needed

## **6. Actions / Scans**

- **Black**: Python code formatter. Does not execute. Only tests.
  Run `black .` locally to execute formatting.
  - [ ] Passed
- **Prettier**: General code formatter. Our use case: MD and yaml mainly.
  Run `npx prettier --write .` locally to execute formatting.
  - [x] Passed
- **Yamllint**: Linting of yaml files.
  - [x] Passed
- **Tests**: Pytest to verify that functionality works as expected.
  - [ ] New tests added
  - [ ] No new tests
  - [x] Passed
- **CodeQL**: Scan for security vulnerabilities, bugs, errors
  - [ ] New alerts: _Go through them and either fix, dismiss och ignore. Add reasoning in items below._
  - [ ] Alerts fixed: _What?_
  - [ ] Alerts ignored / dismissed: _Why?_
  - [x] Passed
- **Trivy**: Security scanner
  - [ ] New alerts: _Go through them and either fix, dismiss och ignore. Add reasoning in items below._
  - [ ] Alerts fixed: _What?_
  - [ ] Alerts ignored / dismissed: _Why?_
  - [x] Passed
- **Snyk**: Security scanner
  - [ ] New alerts: _Go through them and either fix, dismiss och ignore. Add reasoning in items below._
  - [ ] Alerts fixed: _What?_
  - [ ] Alerts ignored / dismissed: _Why?_
  - [x] Passed

<a href="https://gitpod.io/#https://github.com/ScilifelabDataCentre/dds_web/pull/1402"></a>

### CODE REVIEW

File: dds_web/utils.py [hunk 1]
```diff
@@ -292,7 +304,12 @@ def verify_enough_unit_admins(unit_id: str, force_create: bool = False):
 
 def valid_chars_in_username(indata):
     """Check if the username contains only valid characters."""
-    return bool(re.search(r"^[a-zA-Z0-9_\.-]+$", indata))
+    try:
+        contains_only_latin1(indata=indata)
+    except marshmallow.ValidationError:
+        return False
+    else:
+        return bool(re.search(r"^[a-zA-Z0-9_\.-]+$", indata))
```
Comment (i-oden) [inline dds_web/utils.py:312]:
```suggestion
    return bool(re.search(r"^[a-zA-Z0-9_\.-]+$", indata))
```

File: tests/test_user_confirm_invites_and_register.py [hunk 2]
```diff
@@ -172,6 +172,66 @@ def test_register_weak_password(registry_form_data, client):
...
+    form_data["username"] = "user_€"  # € is invalid
+
+    # Request should work
+    response = client.post(
+        tests.DDSEndpoint.USER_NEW,
+        json=form_data,
+        follow_redirects=True,
+        headers=tests.DEFAULT_HEADER,
+    )
+    assert response.status == "200 OK"
```
Comment (valyo) [inline tests/test_user_confirm_invites_and_register.py:191]: is there any particular reason that the status check is done line this at some places:
> assert response.status_code == http.HTTPStatus.OK
and like this at other:
> assert response.status == "200 OK"

I mean, for consistency reasons we should probably stick to one of them

**Review by valyo — CHANGES_REQUESTED** [review]
Apart from that comment, it works and I didn't noticed anything strange

File: tests/test_user_confirm_invites_and_register.py (hunk 2 above)
Comment (i-oden) [reply tests/test_user_confirm_invites_and_register.py:191]: No particular reason, just some copy paste and then changes elsewhere in code. Will fix

**Review by valyo — APPROVED** [review]

**Pull request merged**